app/Http/Controllers/Api/PostsController.php

```diff
@@ -56,20 +64,61 @@ public function store(PostRequest $request)
         $transactionId = incrementTransactionId($request);
 
         // @TODO - Remove. This is temporary. Just hardcoding some params in the request that the client would normally pass. But we assume everything is a photo post from a user at the moment.
-        $request['event_type'] = 'post_photo';
         $request['submission_type'] = 'user';
 
         $signup = $this->signups->get($request['northstar_id'], $request['campaign_id'], $request['campaign_run_id']);
 
+        $updating = ! is_null($signup);
+
         // @TODO - should we eventually throw an error if a signup doesn't exist before a post is created? I create one here because we haven't implemented sending signups to rogue yet, so it will have to create a signup record for all posts.
-        if (is_null($signup)) {
+        if (! $updating) {
             $signup = $this->signups->create($request->all());
+
+            // Send the data to the PostService class which will handle determining
+            // which type of post we are dealing with and which repostitory to use to actually create the post.
+            $post = $this->posts->create($request->all(), $signup->id, $transactionId);
+
+            $code = 200;
+
+            return $this->item($post);
+        } else {
+            $post = $this->posts->update($signup, $request->all(), $transactionId);
+
+            $code = 201;
+
+            if (isset($request['file'])) {
+                return $this->item($post);
+            } else {
+                return $signup;
+            }
+        }
+    }
+
+    /**
+     * Update a post(s)'s status when reviewed.
+     *
+     * @param \Illuminate\Http\Request $request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function reviews(Request $request)
```

Comment:
I sorta feel like reviews should eventually have their own controller, repo, form request class, and stuff since we are considering them like a new api resource. I don't want use to conflate the review endpoint logic with the post logic. Fine to do that in a separate Issue/PR though.